src/shared/calendarspec.c

```diff
@@ -1345,6 +1357,14 @@ static int find_next(const CalendarSpec *spec, struct tm *tm, usec_t *usec) {
                 if (r == 0)
                         continue;
 
+                r = compare_tm(tm, &c);
+                if (r > 0 || (r == 0 && *usec - 1 >= (usec_t) tm_usec)) {
+                        /* We're stuck - advance, let mktime determine DST transition and try again. */
+                        dst = -2;
```

Comment:
i really don't like the overloading here. just add a new boolean local variable?